coalib/results/TextPosition.py

```diff
@@ -15,12 +19,21 @@ def __init__(self, line: (int, None)=None, column: (int, None)=None):
         :param line:        The line in file or None, the first line is 1.
         :param column:      The column indicating the character. The first one
                             in a line is 1.
+        :param text:        The text containing the character.
+        :param position:    Position identifying the index of character in text.
         :raises TypeError:  Raised when line or columns are no integers.
         :raises ValueError: Raised when a column is set but line is None.
         """
         if line is None and column is not None:
             raise ValueError("A column can only be set if a line is set.")
 
+        if line and column and text and not position:
+            position = TextPosition.calc_position(text, line, column)
+
+        if (line and column) is None and (text and position) is not None:
+            line, column = TextPosition.calc_line_col(text, position)
```

Comment:
We may want to do this "on demand". i.e. don't calculate it unless needed.
Basically this is a memory vs speed tradeoff of sorts.
Do we save `text` in memory and later calculate the type which is not given ? (Helps speed wise as we never need to find it sometimes)
or do we calculate it now and save up on the memory ? (Helps memory wise as we dont save it)